lib/probes_.h

```diff
@@ -95,6 +95,7 @@ __attribute__((noinline)) static void h2o_probe_response_header(h2o_req_t *req,
 
 static inline void h2o_probe_log_request(h2o_req_t *req, uint64_t req_index)
 {
+    req->req_index = req_index;
```

Comment:
This looks like a hack, in terms of both assigning the value here and having duplicate state (i.e., having `req_index` here and stream ID in h2 / h3 layer).

How about:
* adding a mandatory callback to `h2o_conn_callbacks_t` that returns the request index given `h2o_req_t *` as an argument, and
* creating `H2O_PROBE_REQUEST` macro like `H2O_PROBE_CONN`, that takes `h2o_req_t *` as the first argument and spits connection ID and request ID to eBPF?

Reply:
That is a great idea, thank you!

Reply:
Done in 21c8f1dfa.

Reply:
Done in cd07df4.